src/pages/ServiceDetails/ServiceInfo.tsx

```diff
@@ -74,6 +74,21 @@ class ServiceInfo extends React.Component<ServiceDetails, ServiceInfoState> {
     return validationChecks;
   }
 
+  differentSources(): string[] {
+    const dependencies = this.props.serviceDetails.dependencies || {};
+
+    let differentDependencies: string[] = [];
```

Comment:
perhaps a Set would also be an appropriate solution here? (if indeed so, then you won't have to check below on line 83 "if not exist" etc.)

Reply:
You are totally right :) Change approached.